lib/PuppeteerSharp/BrowserFetcher.cs

```diff
@@ -347,26 +353,29 @@ internal static Platform GetCurrentPlatform()
 
         internal static string GetExecutablePath()
         {
+            var assembly = typeof(Puppeteer).Assembly;
+            var assemblyName = assembly.GetName().Name + ".dll";
             DirectoryInfo assemblyDirectory = new(AppContext.BaseDirectory);
-            if (!assemblyDirectory.Exists || !File.Exists(Path.Combine(assemblyDirectory.FullName, "PuppeteerSharp.dll")))
+
+            if (!assemblyDirectory.Exists || !File.Exists(Path.Combine(assemblyDirectory.FullName, assemblyName)))
             {
-                string assemblyLocation;
-                var assembly = typeof(Puppeteer).Assembly;
-#pragma warning disable SYSLIB0012 // 'Assembly.CodeBase' is obsolete: 'Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility.
-                if (Uri.TryCreate(assembly.CodeBase, UriKind.Absolute, out var codeBase) && codeBase.IsFile)
-#pragma warning restore SYSLIB0012 // 'Assembly.CodeBase' is obsolete: 'Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility.
-                {
-                    assemblyLocation = codeBase.LocalPath;
-                }
-                else
+                var assemblyLocation = assembly.Location;
+
+                if (string.IsNullOrEmpty(assemblyLocation))
                 {
-                    assemblyLocation = assembly.Location;
+                    var singleFilePublishFilePathForBrowserExecutables = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), nameof(Puppeteer));
```

Comment:
Should the folder name be `PuppeteerSharp`? 

Thoughts?

Reply:
There are a bunch of aspects:
- it is common to use the vendor name for the folder - techies will know what to expect in `PuppeteerSharp`
- shared use of downloaded chrome in multiple puppeteer-sharp consumers - for good or bad
- length vs max path length
- complexity to do magic to get the name out of consuming applications

I would like to keep it simple.

Reply:
Unless I'm mistaken here `nameof(Puppeteer)` = `Puppeteer` 

I think the folder should be named `PuppeteerSharp`.

> I would like to keep it simple.

You've gone way more complex than I was actually meaning 😄